### PR TITLE
Set additional HTTP headers for WCS provider

### PR DESCRIPTION
**Pull request by domi4484**
Same way as done for WMS
Fixes partially #17823 (WFS still missing)

image

### CODE REVIEW

File: src/providers/wcs/qgswcsprovider.h [hunk 1]
```diff
@@ -31,6 +31,7 @@
 #include "qgscoordinatetransform.h"
 #include "qgsogrutils.h"
 #include "qgsapplication.h"
+#include "../wfs/qgsauthorizationsettings.h"
```
Comment (3nids) [inline src/providers/wcs/qgswcsprovider.h:34]: this should be avoided by adding the path in CMakeLists.txt

**Review by 3nids — APPROVED** [review]

File: src/core/auth/qgsauthorizationsettings.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+#include "qgsapplication.h"
+#include "qgshttpheaders.h"
+
+#include <QString>
+#include <QNetworkRequest>
+#include <QNetworkReply>
+
+#define SIP_NO_FILE
+
+struct QgsAuthorizationSettings
```
Comment (nyalldawson) [inline src/core/auth/qgsauthorizationsettings.h:29]: Can you add some docs for the class itself?

Comment (domi4484) [reply src/core/auth/qgsauthorizationsettings.h:35]: @nyalldawson thanks I added some. Do I need to use the macro `CORE_EXPORT` ? The struct is used in some of the providers

**Comment by domi4484** [comment]
@3nids it seems that it can't link the moved struct `QgsAuthorizationSettings`.  Do I need to specify something more in the CMakeLists? The struct is defined header only

**Comment by nyalldawson** [comment]
@domi4484 does it help if you promote it to a class instead of a struct?

**Pull request merged**